File: Project/Windows/Power of God/pSystem/ErrorLogging.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Power_of_God.pSystem
{
    public class ErrorLogging
    {
        private static DateTime _now = DateTime.Now;
        private static readonly string Nl = Environment.NewLine;
        public static void Write(Exception ex)
        {
            if (!Directory.Exists("Logs/")) Directory.CreateDirectory("Logs/");
            var fileLines = new List<string> { "-----Power of God Error Log-----" + Nl };
            try
            {
                fileLines.Add("Exception Message: " + ex.Message);
            }
            catch (Exception)
            {
                fileLines.Add(Nl);
            }
            fileLines.Add(ex.ToString());
            fileLines.Add(Nl + Nl);
            fileLines.Add(ex.StackTrace);
            fileLines.Add("Log wrote at " + _now.ToLongDateString());
            var writingPath = "Logs/" + _now.ToLongDateString() + "." + _now.Hour + "." + _now.Minute + "." + _now.Second + ".txt";
            var fWriter = File.CreateText(writingPath);
            foreach (var f in fileLines)
            {
                fWriter.Write(f + Nl);
            }
            fWriter.Flush();
            fWriter.Close();
        }
    }
}

File: Windows/Updater/ErrorLogging.vb

Imports System.Collections.Generic
Imports System.IO

Public Class ErrorLogging
    Private Shared _now As DateTime = DateTime.Now
    Private Shared ReadOnly Nl As String = Environment.NewLine
    Public Shared Sub Write(ex As Exception)
        If Not Directory.Exists("Logs/") Then
            Directory.CreateDirectory("Logs/")
        End If
        Dim fileLines = New List(Of String)() From {
                Convert.ToString("-----Power of God Error Log (Updater)-----") & Nl
                }
        Try
            fileLines.Add("Exception Message: " + ex.Message)
        Catch generatedExceptionName As Exception
            fileLines.Add(Nl)
        End Try
        fileLines.Add(ex.ToString())
        fileLines.Add(Nl & Nl)
        fileLines.Add(ex.StackTrace)
        fileLines.Add("Log wrote at " + _now.ToLongDateString())
        Dim writingPath = "Logs/" & "Updater" & _now.ToLongDateString() & "." & _now.Hour & "." & _now.Minute & "." + _now.Second & ".txt"
        Dim fWriter = File.CreateText(writingPath)

        For Each f In fileLines
            fWriter.Write(Convert.ToString(f) & Nl)

        Next
        fWriter.Flush()
        fWriter.Close()
    End Sub
End Class

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
